packages/block-library/src/navigation/edit/use-convert-classic-menu-to-block-menu.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * WordPress dependencies
+ */
+import { useRegistry } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { useState, useCallback } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useCreateNavigationMenu from './use-create-navigation-menu';
+import menuItemsToBlocks from '../menu-items-to-blocks';
+
+const CLASSIC_MENU_CONVERSION_SUCCESS = 'success';
+const CLASSIC_MENU_CONVERSION_ERROR = 'error';
+const CLASSIC_MENU_CONVERSION_PENDING = 'pending';
+
+function useConvertClassicToBlockMenu( clientId ) {
+	const createNavigationMenu = useCreateNavigationMenu( clientId );
+	const registry = useRegistry();
+
+	const [ status, setStatus ] = useState( 'idle' );
+	const [ value, setValue ] = useState( null );
+	const [ error, setError ] = useState( null );
+
+	async function convertClassicMenuToBlockMenu( menuId, menuName ) {
+		let navigationMenu;
+		let classicMenuItems;
+
+		const menuItemsParameters = {
+			menus: menuId,
+			per_page: -1,
+			context: 'view',
+		};
+
+		const fetchError = new Error(
+			sprintf(
+				// translators: %s: the name of a menu (e.g. Header navigation).
+				__( `Unable to fetch classic menu "%s" from API.` ),
+				menuName
+			),
+			{
+				menuId,
+				menuName,
+			}
+		);
+
+		// 1. Fetch the classic Menu items.
+		try {
+			classicMenuItems = await registry
+				.resolveSelect( coreStore )
+				.getMenuItems( menuItemsParameters );
+		} catch ( e ) {
+			throw fetchError;
+		}
+
+		// Handle offline response which resolves to `null`.
+		if ( classicMenuItems === null ) {
+			throw fetchError;
+		}
+
+		// 2. Convert the classic items into blocks.
+		const { innerBlocks } = menuItemsToBlocks( classicMenuItems );
+
+		// 3. Create the `wp_navigation` Post with the blocks.
+		try {
+			navigationMenu = await createNavigationMenu(
+				menuName,
+				innerBlocks
+			);
+		} catch ( e ) {
+			throw new Error(
+				sprintf(
+					// translators: %s: the name of a menu (e.g. Header navigation).
+					__( `Unable to create Navigation Menu "%s".` ),
+					menuName
+				),
+				{
+					menuId,
+					menuName,
+				}
```

Comment:
This could be deleted, too

Reply:
Or better, replaced with { cause: e }

Reply:
I did see this `cause` and was thinking "I'll come back to that". Guess what...I forgot! Lesson - don't leave issues/problems undocumented as you come across them - you will forget 😄 

Thanks for catching the silly usage of `options`. I've now updated to pass the original error as `cause`.